packages/components/src/menu-item/index.tsx

```diff
@@ -32,18 +33,32 @@ function UnforwardedMenuItem(
 		isSelected,
 		role = 'menuitem',
 		suffix,
-		...buttonProps
+		...restProps
 	} = props;
 
 	className = clsx( 'components-menu-item__button', className );
 
+	const buttonProps = { ...restProps };
 	if ( info ) {
+		const labelId = `menu-item-label-${ uuidv4() }`;
+		const descriptionId = `menu-item-description-${ uuidv4() }`;
+
 		children = (
 			<span className="components-menu-item__info-wrapper">
-				<span className="components-menu-item__item">{ children }</span>
-				<span className="components-menu-item__info">{ info }</span>
+				<span id={ labelId } className="components-menu-item__item">
+					{ children }
+				</span>
+				<span className="components-menu-item__info" aria-hidden="true">
+					{ info }
+				</span>
+				<span id={ descriptionId } className="screen-reader-text">
+					{ info }
+				</span>
```

Comment:
I would move the `info` span outside of the children markup entirely and use a `hidden` html attribute instead of `className="screen-reader-text"`. It can eveb be placed outside of the Item. This way the markup would be cleaner and there wouldn't be the need to use `labelId` and `aria-labelledby` because the labeling would simply use the content.